spacy/tests/test_parallel.py

```diff
@@ -0,0 +1,238 @@
+        load_project_config(d)
+
+
+@pytest.mark.parametrize(
+    "workflows",
+    [
+        {"all": ["command1", {"parallel": ["command2"]}, "command3"]},
+        {"all": ["command1", {"parallel": ["command2", "command4"]}]},
+        {"all": ["command1", {"parallel": ["command2", "command2"]}]},
+        {"all": ["command1", {"serial": ["command2", "command3"]}]},
+    ],
+)
+def test_project_config_multiprocessing_bad_case_workflows(workflows):
+    project = {
+        "commands": [
+            {"name": "command1", "script": ["echo", "command1"]},
+            {"name": "command2", "script": ["echo", "command2"]},
+            {"name": "command3", "script": ["echo", "command3"]},
+        ],
+        "workflows": workflows,
+    }
+    with make_tempdir() as d:
+        srsly.write_yaml(d / "project.yml", project)
+        with pytest.raises(SystemExit):
+            load_project_config(d)
+
+
+@pytest.mark.parametrize("max_parallel_processes", [-1, 0, 1])
+def test_project_config_multiprocessing_max_processes_bad_case(max_parallel_processes):
+    with make_tempdir() as d:
+        project = {
+            "max_parallel_processes": max_parallel_processes,
+            "commands": [
+                {
+                    "name": "commandA",
+                    "script": [" ".join(("touch", os.sep.join((str(d), "A"))))],
+                },
+                {
+                    "name": "commandB",
+                    "script": [" ".join(("touch", os.sep.join((str(d), "B"))))],
+                },
+                {
+                    "name": "commandC",
+                    "script": [" ".join(("touch", os.sep.join((str(d), "C"))))],
+                },
+            ],
+            "workflows": {"all": [{"parallel": ["commandA", "commandB", "commandC"]}]},
+        }
+        with make_tempdir() as d:
+            srsly.write_yaml(d / "project.yml", project)
+            with pytest.raises(SystemExit):
+                load_project_config(d)
+
+
+def test_project_run_multiprocessing_good_case():
+    with make_tempdir() as d:
+
+        pscript = """
+import sys, os
+from time import sleep
+
+_, d_path, in_filename, out_filename, first_flag_filename, second_flag_filename = sys.argv
+with open(os.sep.join((d_path, out_filename)), 'w') as out_file:
+    out_file.write("")
+while True:
+    if os.path.exists(os.sep.join((d_path, in_filename))):
+        break
+    sleep(0.1)
+if not os.path.exists(os.sep.join((d_path, first_flag_filename))):    
+    with open(os.sep.join((d_path, first_flag_filename)), 'w') as first_flag_file:
+        first_flag_file.write("")
+else: # should never happen because of skipping
+    with open(os.sep.join((d_path, second_flag_filename)), 'w') as second_flag_file:
+        second_flag_file.write("")
+        """
+
+        pscript_loc = os.sep.join((str(d), "pscript.py"))
+        with open(pscript_loc, "w") as pscript_file:
+            pscript_file.write(pscript)
```

Comment:
Think it'd be better to do this as temporary file.

Reply:
Is that necessary given that we are already writing to a temporary directory? Does a temporary file have additional benefits in this situation?

Reply:
Ah, I completely missed you're doing `make_tempdir()` anyway :slightly_smiling_face: